### ISSUE TITLE
P_RoleRule confusion

### ISSUE THREAD

**Issue by WolframKahl** [AmpersandTarski/Ampersand#265]
In ```src/Database/Design/Ampersand/Core/ParseTree.hs``` I find:
```
 -- | A RoleRule r means that a role called 'mRoles r' must maintain the process rule called 'mRules r'
data P_RoleRule
   = Maintain
     { mPos :: Origin      -- ^ position in the Ampersand script
     , mRoles :: [Role]    -- ^ name of a role
     , mRules :: [String]  -- ^ name of a Rule
     }
```
The field name ```nRoles``` is a natural-language plural, its type ```[Role]``` is naturally interpreted as a Haskell-plural of ```Role```, but the comment says that this list of ```Role``` elements is a single name of a single role?
(Almost the same for ```mRules```.) Are these multi-part names or not?
The comment before the data-declaration also suggests yes, so if they are, I would strongly recommend to change these field names. If not, all comments need thorough rewriting.

(Is creating an issue the right way to point out things like this?)

Wolfram

**Comment by hanjoosten** [comment]
The comment has clearly not been updated. The Haskell type clearly states that there is a list of Roles. In the past, the P_RoleRule datatype was about two strings that contained the name of the role and another that contained the name of a rule. 
It would be easy to fix this particular comment. However, I fear that the comments in the data types (specially the ones declared in src/Database/Design/Ampersand/Core/ParseTree.hs and in src/Database/Design/Ampersand/Core/AbstractSyntaxTree.hs ) should be revitalized. The Haddock documentation of Ampersand isn't very accurate, The type system is. GHC 8.0 will have a feature called [OverloadedLabels](https://ghc.haskell.org/trac/ghc/wiki/Records/OverloadedRecordFields/OverloadedLabels). This will enable us to have better names of the labels used in data types. We hope to be able to use the same names for labels as we use in the FormalAmpersand.adl files. If we choose to do so, we need to take the documentation of all datatypes into account as well.

**Comment by stefjoosten** [comment]
Thanks, @WolframKahl for this issue. Yes, creating an issue is the right way to point out things like this.

the reason: Even in the worst-case, where your remark would be ignored, there still must be someone  to take a decision to close the issue. So even ignoring becomes explicit.

**Comment by sjcjoosten** [comment]
I've updated some comments in the parser to close this issue. I've also experimented with OverloadedLabels: all positions in the parsetree are now called "pos", which seems to work reasonably well, although it's still beneficial to use "origin", since the disambiguation of "pos" doesn't work too well (for now). We can discuss whether we want to use this trick throughout the Ampersand code in a separate issue if someone would be in favor of such a refactoring.
